src/repeated_game.jl

```diff
@@ -0,0 +1,339 @@
+"""
+unpack(rpd::RepeatedGame) = (rpd.sg, rpd.δ)
+
+# Flow utility in terms of the players actions
+flow_u_1(rpd, a1::Int, a2::Int) = (1-rpd.δ)*rpd.sg.players[1].payoff_array[a1, a2]
+flow_u_2(rpd, a1::Int, a2::Int) = (1-rpd.δ)*rpd.sg.players[2].payoff_array[a2, a1]
+flow_u(rpd, a1::Int, a2::Int) = [flow_u_1(rpd, a1, a2), flow_u_2(rpd, a1, a2)]
+
+# Computes each players best deviation given an opponent's action
+best_dev_i(rpd, i::Int, aj::Int) = indmax(rpd.sg.players[i].payoff_array[:, aj])
+best_dev_1(rpd, a2::Int) = best_dev_i(rpd, 1, a2)
+best_dev_2(rpd, a1::Int) = best_dev_i(rpd, 2, a1)
+
+# Computes the payoff of the best deviation
+best_dev_payoff_i(rpd, i::Int, aj::Int) = (1-rpd.δ)*maximum(rpd.sg.players[i].payoff_array[:, aj])
+best_dev_payoff_1(rpd, a2::Int) = (1-rpd.δ)*maximum(rpd.sg.players[1].payoff_array[:, a2])
+best_dev_payoff_2(rpd, a1::Int) = (1-rpd.δ)*maximum(rpd.sg.players[2].payoff_array[:, a1])
+
+"""
+Creates the unit circle of continuation value points starting at origin `o`
+with radius `r`. Additionally, this function crops any points that go
+beyond the bounds permitted by the game -- For example, if the max payoff
+for player 1 is 5 then any point that awards player 1 higher than 5
+continuation utility is cropped to 5.
+"""
+function create_unitcircle_points{T<:Real}(rpd::RepeatedGame{2, T}, nH::Int,
+                                           o::Vector{Float64}, r::Float64)
+    # First create unit circle
+    H = unitcircle(nH)
+    HT = H'
+
+    # Choose origin and radius for big approximation
+    p1_min, p1_max = extrema(rpd.sg.players[1].payoff_array)
+    p2_min, p2_max = extrema(rpd.sg.players[2].payoff_array)
+    Z = Array(Float64, 2, nH)
+    for i=1:nH
+        # We know that players can ever get worse than their
+        # lowest punishment, so ignore anything below that
+        Z[1, i] = min(max(o[1] + r*HT[1, i], p1_min), p1_max)
+        Z[2, i] = min(max(o[2] + r*HT[2, i], p2_min), p2_max)
+    end
+
+    # Corresponding hyperplane levels
+    C = squeeze(sum(HT .* Z, 1), 1)
+
+    return C, H, Z
+end
+
+function create_unitcircle_points(rpd::RepeatedGame, nH::Int)
+
+    # Choose the origin to be mean of max and min payoffs
+    p1_min, p1_max = extrema(rpd.sg.players[1].payoff_array)
+    p2_min, p2_max = extrema(rpd.sg.players[2].payoff_array)
+
+    o = [(p1_min + p1_max)/2.0, (p2_min + p2_max)/2.0]
+    r1 = max((p1_max - o[1])^2, (o[1] - p1_min)^2)
+    r2 = max((p2_max - o[2])^2, (o[2] - p2_min)^2)
+    r = sqrt(r1 + r2)
+
+    return create_unitcircle_points(rpd, nH, o, r)
+end
+
+#
+# Linear Programming Functions
+#
+"""
+Initialize matrices for the linear programming problems. It sets up the A
+matrix since it never changes, but only allocates space for b and c since
+they will be filled repeatedly on the iterations
+
+We add nH slack variables (which will be constrained to be positive) to
+deal with inequalities associated with Ax \leq b.
+
+min c ⋅ x
+    Ax = b
```

Comment:
How come you convert constraints to equalities? Doesn't `ClpSolver` accept `'<'` (as in the case in `worst_value_i`)?